File: apps/flights-service-admin/src/api/flight/Flight.ts
import { Airline } from "../airline/Airline";
import { Airport } from "../airport/Airport";

export type Flight = {
  airline?: Airline | null;
  arrivalAirport?: Airline | null;
  arrivalTime: Date | null;
  availableSeats: number | null;
  createdAt: Date;
  departureAirport?: Airport | null;
  departureTime: Date | null;
  flightNumber: string | null;
  id: string;
  updatedAt: Date;
};

File: apps/flight-catalog/src/APIs/Flight/Dtos/Flight.cs
namespace FlightCatalog.APIs.Dtos;

public class Flight
{
    public string? Airline { get; set; }

    public DateTime? ArrivalTime { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime? DepartureTime { get; set; }

    public string? FlightNumber { get; set; }

    public string Id { get; set; }

    public string? Route { get; set; }

    public List<string>? Schedules { get; set; }

    public List<string>? Tickets { get; set; }

    public DateTime UpdatedAt { get; set; }
}
